src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -2280,7 +2234,7 @@ test_cursor_implicit_session (void *ctx)
    BSON_ASSERT (cursor->client_session);
    BSON_ASSERT (!cursor->explicit_session);
    bson_copy_to (&cursor->client_session->server_session->lsid, &find_lsid);
-   ASSERT_POOL_SIZE (topology, 0);
+   BSON_ASSERT (mongoc_ts_pool_size (topology->session_pool) == 0);
```

Comment:
I suggest using `ASSERT_CMPINT`, which prints the values being compared on failure.